src/vs/workbench/contrib/tasks/common/taskConfiguration.ts

```diff
@@ -1470,15 +1465,15 @@ namespace ConfiguringTask {
 		let taskSource: Tasks.FileBasedTaskSource;
 		switch (source) {
 			case TaskConfigSource.User: {
-				taskSource = Object.assign({} as Tasks.IUserTaskSource, partialSource, { kind: Tasks.TaskSourceKind.User, config: configElement });
+				taskSource = Object.assign({}, { kind: Tasks.TaskSourceKind.User, config: configElement, label }) satisfies Tasks.IUserTaskSource;
```

Comment:
It might be clearer just to use spreads here:

```ts
=  { ...partialSource, kind: ..., config: ...,  }
```

Reply:
I think the `partialSource` thing is confusing when it's really just a label. I'd rather go with this.